### PR TITLE
Fixes bug preventing usage of feedRangeFilter with change feed

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
We had too aggressive validation in CosmosPartitionPlanner that was uncovered when using readStream in combination with a FeedRangeFilter that only filters on single logical partition. Relaxed the validation and added test coverage.

Fixes#28628

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala [hunk 1]
```diff
@@ -396,25 +396,40 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
         Some(range)
       })
 
+    val result = new ArrayBuffer[PartitionMetadata]
     orderedRanges
-      .map(range => {
-        while (!SparkBridgeImplementationInternal.doRangesOverlap(range, startTokens(startTokensIndex)._1)) {
+      .foreach(range => {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala:401]: If there is one range we can not find any overlapping, what would happen? 

Previously, an exception is going to throw?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala:401]: It is fine to have one/multiple ranges without overlapping as long as there is at least one final range - can happen due to feed range filtering - but also multiple splits/merges between micro batch runs

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala:401]: And if there is no range with overlap there is still an assert

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala [hunk 2]
```diff
@@ -396,25 +396,40 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
...
-        val startLsn: Long = startTokens(startTokensIndex)._2
-        latestTokens(latestTokensIndex).cloneForSubRange(range, startLsn)
+        if (startTokensIndex < startTokens.length &&
+          latestTokensIndex < latestTokens.length) {
+
+          val startLsn: Long = startTokens(startTokensIndex)._2
+          val newToken = latestTokens(latestTokensIndex).cloneForSubRange(range, startLsn)
+          result.append(newToken)
+        } else {
+          startTokensIndex = initialStartTokensIndex
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala:425]: If there is one range we can not find any overlapping, what would happen?

Previously, an exception is going to throw?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Pull request merged**